language-server/vscoqtop/lspManager.ml

```diff
@@ -257,7 +257,15 @@ let coqtopUpdateProofView ~id params =
   let loc = params |> member "position" |> parse_loc in
   let st = Hashtbl.find states uri in
   match Dm.DocumentManager.get_proof st loc with
-  | None -> ()
+  (* Send an "empty" response to notify about not being in a proof *)
+  (* TODO: Selectively updates based on prior state (should not alert when already not in proof) *)
```

Comment:
Actually, I disagree with the `TODO` (i.e. I think your current approach is fine) :)
The protocol should be stateless, it is the job of the UI to be smart about how it notifies IMHO.

Reply:
That is a fair point, I'll add it to my review fixes.